src/use-provide-context.ts

```diff
@@ -0,0 +1,78 @@
+import { Context, ContextDetail } from "./create-context";
+import { Hook, hook } from "./hook";
+import { State } from "./state";
+import { contextEvent } from "./symbols";
+
+/**
+ * @function
+ * @template T
+ * @param  {Context} Context Context to provide a value for
+ * @param  {T} value the current value
+ * @param  {unknown[]} values dependencies to the value update
+ * @return void
+ */
+export const useProvideContext = hook(
+  class<T> extends Hook<[Context<T>, T, unknown[]], void, Element> {
+    Context!: Context<T>;
```

Comment:
should this be
```suggestion
    context!: Context<T>;
```

also, and this is a matter of taste, you could use [parameter properties](https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties) for brevity

Reply:
@bennypowers thanks!